modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java

```diff
@@ -323,7 +325,13 @@ public AMQConnection(ConnectionURL connectionURL, SSLConfiguration sslConfig) th
             _useLegacyMapMessageFormat = Boolean.getBoolean(ClientProperties.USE_LEGACY_MAP_MESSAGE_FORMAT);
         }
 
-        String amqpVersion = System.getProperty((ClientProperties.AMQP_VERSION), "0-10");
+        // Requires permission java.util.PropertyPermission "qpid.amqp.version", "write"
+        String amqpVersion = (String) AccessController.doPrivileged(new PrivilegedAction() {
```

Comment:
Can't we use the generic version of PrivilegedAction[1] so that we don't have to cast to String?

[1] http://docs.oracle.com/javase/7/docs/api/java/security/PrivilegedAction.html